src/index.js

```diff
@@ -105,9 +105,9 @@ async function main(event) {
 
     for (const record of event.Records) {
       const partialStart = process.hrtime.bigint()
-      const [, bucket, key] = record.body.match(/([^/]+)\/(.+)/)
+      const [, bucketRegion, bucketName, key] = record.body.match(/([^/]+)\/(.+)\/(.+)/)
```

Comment:
I would use `/([^/]+)\/([^/]+)\/(.+)/` to make sure no slashes are mistakenly used in region or name.